pbft/pbft.cpp

```diff
@@ -588,6 +624,24 @@ pbft::do_committed(const std::shared_ptr<pbft_operation>& op)
             // get rid of all other previous configs, except for currently active one
             this->configurations.remove_prior_to(config.get_hash());
         }
+
```

Comment:
Should move this to after `op->advance_operation_stage(pbft_operation_stage::execute);`; it' has a relevant sanity check

Reply:
Will do. Side note though - seems like an easy attack for someone to send us a commit for an operation that isn't ready to commit and we'll throw an exception and exit...

Reply:
Actually I guess it's not that easy to get into that state.

Reply:
Regardless of the likelihood of it happening here, you've got a point about that issue in general; I'm vaguely aware of several places where a bad peer could cause crashes by sending bad data. I'll make a backlog task to go over that.